modules/dream2nix/rust-cargo-lock/interface.nix

```diff
@@ -15,7 +15,8 @@ in {
     source = {
       type = t.either t.path t.package;
       description = "Source of the package";
-      default = config.mkDerivation.src;
+      # pull default either from rust-crane source or mkDerivation source (mkDerivation is what buildRustPackage uses)
+      default = config.rust-crane.source or config.mkDerivation.src;
```

Comment:
default values that depend on other config values might be problematic when generating documentation. Could you move this to rust-cargo-lock/default.nix and use `mkDefault` instead?